clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -198,7 +198,7 @@ private static FilterResult filterTo(TopicPartition partition, Iterable<MutableR
             if (!retainedRecords.isEmpty()) {
                 if (writeOriginalBatch) {
                     batch.writeTo(bufferOutputStream);
-                    filterResult.updateRetainedBatchMetadata(batch, retainedRecords.size(), false);
+                    filterResult.updateRetainedBatchMetadata(batch, batch.baseOffset(), retainedRecords.size(), false);
```

Comment:
nit: why do we need to pass the base offset as a separate parameter? We are already passing `batch`.